### DISCUSSION TITLE
Memory fragmentation problem

### DISCUSSION THREAD

**Discussion by Marcus4242** [schreibfaul1/ESP32-audioI2S#478] [Q&A]
Hi,
I have a program that allows playing mp3 files from SD card via Bluetooth BLE commands.
However, I'm quickly running into heap memory fragmentation problems and therefore the mp3 decoder can't allocate enough memory and simply stops playing my files.
Is there any way I can avoid this? Is it possible to allocate enough memory right from the start so it stays static and avoids fragmentation? Or are there any other ideas? My esp doesn't have psram, but I ordered one to see if that might help, too.

**Comment by FreeLANMan** [comment]
Test different versions of the lib. 
To see fewer messages like 'Guru Meditation Error: Core 1 panic'ed'
